gpiozero/boards.py

```diff
@@ -2239,3 +2239,104 @@ def off(self):
         """
         self.buzzer.value = None
         super(JamHat, self).off()
+
+class Pibrella(CompositeOutputDevice):
+    """
+    Extends :class:`CompositeOutputDevice` for the Cyntech/Pimoroni `Pibrella`_
+    board.
+
+    The Pibrella board comprises 3 LEDs, a button, a tonal buzzer, four general
+    purpose input channels, and four general purpose output channels (with LEDs).
+
+    This class exposes the LEDs, button and buzzer.
+
+    Usage::
+
+        from gpiozero import Pibrella
+
+        pb = Pibrella()
+
+        pb.button.wait_for_press()
+        pb.lights.on()
+        pb.buzzer.play('A4')
+        pb.off()
+
+    The four input and output channels are not provided by this class. Instead,
+    you can create GPIO Zero devices using these pins::
+
+        from gpiozero import Pibrella, LED, Button
+
+        pb = Pibrella()
+        btn = Button(pb.inputs.a)
+        led = LED(pb.outputs.e)
+
+        btn.when_pressed = led.on
+
+    :param bool pwm:
+        If :data:`True` (the default), construct :class: PWMLED instances to
+        represent each LED on the board. If :data:`False`, construct regular
+        :class:`LED` instances.
+
+    :type pin_factory: Factory or None
+    :param pin_factory:
+        See :doc:`api_pins` for more information (this is an advanced feature
+        which most users can ignore).
+
+    .. _Pibrella: http://www.pibrella.com/
+
+    .. attribute:: lights
+
+        :class:`TrafficLights` instance representing the three LEDs
+
+        .. attribute:: red, amber, green
+
+            :class:`LED` or :class:`PWMLED` instances representing the red,
+            amber, and green LEDs
+
+    .. attribute:: button
+
+        The red :class:`Button` objects on the Pibrella
+
+    .. attribute:: buzzer
+
+        A :class:`TonalBuzzer` object representing the buzzer
+
+    .. attribute:: inputs
+
+        A namedtuple of the input pin numbers
+
+        .. attribute:: a, b, c, d
+
+    .. attribute:: outputs
+
+        A namedtuple of the output pin numbers
+
+        .. attribute:: e, f, g, h
+    """
+    def __init__(self, pwm=True, pin_factory=None):
+        super(Pibrella, self).__init__(
+            lights=TrafficLights(red=27, amber=17, green=4, pwm=pwm,
+                              pin_factory=pin_factory),
+            button=Button(11, pull_up=False, pin_factory=pin_factory),
+            buzzer=TonalBuzzer(18, pin_factory=pin_factory),
+            _order=('lights', 'buzzer', 'button'),
+            pin_factory=pin_factory
+        )
+        InputPins = namedtuple('InputPins', ['a', 'b', 'c', 'd'])
+        OutputPins = namedtuple('OutputPins', ['e', 'f', 'g', 'h'])
+        self.inputs = InputPins(a=9, b=7, c=8, d=10)
```

Comment:
I _think_ you might be able to create the actual `Pin` objects here (whilst using the appropriate `pin_factory`), rather than just storing the gpio numbers?

Reply:
I did think of that, but couldn't remember if a Pin object is a valid pin parameter any more.

Reply:
ping @waveform80

Reply:
Looks like it's not. You need to pass in a number or string, and `pin_factory` (optional) so I don't think `pb.outputs.e` being an instance of `RPiGPIOPin` is useful, as you can't create a device object with it.

Reply:
Fair enough. The "pins" stuff has gone through so many changes, it's hard to remember what is and isn't supported. And I just tried having a quick look at the sourcecode just now, and there's so many abstract classes, factory classes, super method calls, etc. etc. that I'm still not able to follow exactly what's going on :-)